ipatests/test_integration/test_hbac_functional.py

```diff
@@ -0,0 +1,3612 @@
+        """Install and initial setup"""
+        # Install FTP
+        tasks.install_packages(cls.master, ['vsftpd'])
+        tasks.install_packages(cls.clients[0], ['ftp'])
+        tasks.install_packages(cls.clients[1], ['ftp'])
+        if cls.domain_level is not None:
+            domain_level = cls.domain_level
+        else:
+            domain_level = cls.master.config.domain_level
+        tasks.install_topo(cls.topology,
+                           cls.master, [],
+                           cls.clients, domain_level,
+                           clients_extra_args=('--mkhomedir',))
+        tasks.kinit_admin(cls.master)
+        # Setup common test users
+        cls.setup_common_users()
+
+    @classmethod
+    def setup_common_users(cls):
+        """Create common test users"""
+        tasks.kinit_admin(cls.master)
+        for user in [cls.USER_1, cls.USER_2, cls.USER_3]:
+            # Create the test active user's
+            tasks.create_active_user(
+                cls.master, user, password=cls.USER_PASSWORD, extra_args=[
+                    '--homedir', '/home/{}'.format(user)]
+            )
+            # Set home directory ownership and permissions
+            home_dir = f'/home/{user}'
+            cls.master.run_command(['mkdir', '-p', home_dir])
+            cls.master.run_command(['chown', f'{user}:{user}', home_dir])
+            cls.master.run_command(['chmod', '755', home_dir])
+        tasks.clear_sssd_cache(cls.master)
+
+    def ssh_auth_success(self, user, password, host, source_host=None):
+        """
+        Test SSH authentication success
+        Args:
+            user: Username to authenticate
+            password: Password for authentication
+            host: Host to connect to (target host)
+            source_host: Host to run command from (optional,
+                         defaults to target host)
+        Returns:
+            True if authentication succeeds, False otherwise
+        """
+        if source_host is None:
+            source_host = host
+
+        result = source_host.run_command([
+            'sshpass', '-p', password, 'ssh',
+            '-o', 'StrictHostKeyChecking=no',
+            '-o', 'PasswordAuthentication=yes',
+            '-l', user, host.hostname, 'echo login successful'
+        ], raiseonerr=False)
+
+        if result.returncode == 0 and 'login successful' in result.stdout_text:
+            return True
+        return False
+
+    def ssh_auth_failure(self, user, password, host, source_host=None):
+        """
+        Test SSH authentication failure
+        Args:
+            user: Username to authenticate
+            password: Password for authentication
+            host: Host to connect to (target host)
+            source_host: Host to run command from (optional,
+                         defaults to target host)
+        Returns:
+            True if authentication fails as expected, False otherwise
+        """
+        if source_host is None:
```

Comment:
You can simplify this method with `return not self.ssh_auth_success(user, password, host, source_host)`